packages/next-swc/crates/next-core/src/next_app/bloom_filter.rs

```diff
@@ -0,0 +1,186 @@
+use std::{collections::HashSet, f64::consts::LN_2};
```

Comment:
Why do we need this to be duplicated instead of re-using the existing `createClientRouterFilter` handling?

Reply:
We currently don't have an easy way of calling javascript from rust without spawning a new node.js process and having this in rust we can use turbopack's HMR out of the box

I do agree it's suboptimal and we should look into passing all the defines from next.js to rust soon

Reply:
I think we should pass it via env instead of maintaining two versions, ideally we just use the JavaScript version as that's what the client is going to be expecting. 

Even if we don't do define env we could pass it to rust via a normal env variable and turbopack expect it be present if needed.

Reply:
still requires some bigger changes on the js and rust side to get it to that point

for example rust currently doesn't inject env variables directly, it adds them to the `process.env` object

it also does the same filtering as next.js only including vars prefixed with `NEXT_PUBLIC_` or as a part of `next_config.env`k

Reply:
This sounds off as `nextConfig.env` needs to inline all values regardless of `NEXT_PUBLIC_` otherwise it wouldn't work. Last this was discussed with @sokra that field should just work